src/ebfloeseg/bbox.py

```diff
@@ -57,10 +57,19 @@ def convert(self, value, param=None, ctx=None):
             >>> BoundingBoxParser.convert(BoundingBox(-2334051, -414387, -1127689, 757861,))
             BoundingBox(x1=-2334051, y1=-414387, x2=-1127689, y2=757861)
 
+            >>> BoundingBoxParser.convert((1, 2, 3, 4))
+            Traceback (most recent call last):
+            ...
+            NotImplementedError: (1, 2, 3, 4), of type <class 'tuple'>, can't be parsed as a BoundingBox
+
+
         """
-        if isinstance(value, str):
+        if isinstance(value, BoundingBox):
+            pass  # the value is the correct type
+        elif isinstance(value, str):
             raw_value = ast.literal_eval(value)
             value = BoundingBox(*raw_value)
-        elif isinstance(value, BoundingBox):
-            pass
+        else: 
+            msg = "%s, of type %s, can't be parsed as a BoundingBox" % (value, type(value))
+            raise NotImplementedError(msg)
```

Comment:
If you are expecting `value` to be of type `BoundingBox` or `str`, why not add a type annotation to the method signature, which will clearly communicate that intend, and just handle the conversion to that type if a string is passed?
```ptyhon
def convert(self, value:BoundingBox | str, param=None, ctx=None):
```

The string might not have the right format after all so parsing it might not succeed.

It feels strange to me to have an if statement with a pass in it.